packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -408,7 +408,22 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
             )
           }
           needQueryInjectHelper = true
-          str().overwrite(start, end, `__vite__injectQuery(${url}, 'import')`)
+          if (
+            !/^('.*'|".*"|`.*`)$/.test(url) ||
+            (!isJSRequest(cleanUrl(url.slice(1, -1))) &&
```

Comment:
We already have the same pattern here
```js
function markExplicitImport(url: string) {
  if (!isJSRequest(cleanUrl(url)) && !isCSSRequest(url)) {
    return injectQuery(url, 'import')
  }
  return url
}
```

I think we should refactor that check into 
```js
function isExplicitImportRequired(url) {
  return !isJSRequest(cleanUrl(url)) && !isCSSRequest(url)
}
function markExplicitImport(url: string) {
  if (isExplicitImportRequired(url)) {
    return injectQuery(url, 'import')
  }
  return url
}
```
So you can then use it for your change set
```js
if ( !/^('.*'|".*"|`.*`)$/.test(url) || isExplicitImportRequired(url.slice(1, -1)) ) {
```